test/inductor/test_mkldnn_pattern_matcher.py

```diff
@@ -2605,6 +2605,34 @@ def matcher_check_fn():
                 rtol=0.07,
             )
 
+    @skipIfNoDynamoSupport
+    def test_woq_int8_v2(self):
+        class M(torch.nn.Module):
+            def forward(self, x, weight, scales):
+                return (
+                    torch.matmul(x, weight.permute([1, 0]).to(dtype=x.dtype)) * scales
+                )
+
+        mod = M().eval()
+        x_shape = (1, 256)
+        w_shape = (12, 256)
+        s_shape = 12
+        x = torch.randn(x_shape, dtype=torch.bfloat16)
+        w = torch.randint(-128, 127, w_shape, dtype=torch.int8)
+        s = torch.randn(s_shape, dtype=torch.float32)
```

Comment:
does the scale have to be fp32, or it can be of the same type as the input tensor (bf16) as well?